runtime/rasdump/javadump.cpp

```diff
@@ -1498,7 +1505,19 @@ JavaCoreDumpWriter::writeMemorySection(void)
 static UDATA
 countMemoryCategoriesCallback (U_32 categoryCode, const char * categoryName, UDATA liveBytes, UDATA liveAllocations, BOOLEAN isRoot, U_32 parentCategoryCode, OMRMemCategoryWalkState * state)
 {
-	(*(I_32 *)state->userData1)++;
+	(*(U_32 *)state->userData1)++;
+	
+	memcategory_max_indexes *max_indexes = (memcategory_max_indexes *)state->userData2;
+	if (categoryCode > OMRMEM_LANGUAGE_CATEGORY_LIMIT) {
+		U_32 omrCode = OMRMEM_OMR_CATEGORY_INDEX_FROM_CODE(categoryCode);
+		if (omrCode > max_indexes->omrMaxIndex) {
+			max_indexes->omrMaxIndex = omrCode;
```

Comment:
I this this is easier to read if the comparison is reversed (same for `languageMaxIndex` below):
```
    if (max_indexes->omrMaxIndex < omrCode) {
        max_indexes->omrMaxIndex = omrCode;
    }
```

Reply:
updated